src/backend/main.ts

```diff
@@ -76,34 +71,36 @@ export function prepareResponseForStorage(
   uuid: () => string = uuidV4,
   timestamp = Date.now,
 ): Promise<BackendResponseModelT> {
-  return Promise.resolve(secretPepper).then(secretPepper => {
-    const { readPromise, writePromise } = performAbuseDetection(dynamoDb, fingerprint, val => hash(val, secretPepper));
-    writePromise // we don't really care about the write operation here - it can finish on its own (we only need to handle its possible failure; if it keeps failing we want to know)
-      .catch(err => console.log(`Error: Couldn't write abuse score for response (caused by\n${err}\n)`));
-    return readPromise // we only care about the read operation
-      .catch(
-        (err): AbuseScore => {
-          console.log(`Error: Couldn't read abuse score for response; marking with error code (caused by\n${err}\n)`);
-          return ABUSE_SCORE_ERROR;
-        },
-      )
-      .then(abuse_score => {
-        const meta = {
-          response_id: uuid(),
-          participant_id: hash(hash(response.participant_id, knownPepper), secretPepper), // to preserve privacy, hash the participant_id before storing it, so after opening up the dataset, malicious actors can't submit more responses that pretend to belong to a previous participant
-          timestamp: new Date(timestamp()) // for security, don't trust browser clock, as it may be wrong or fraudulent
-            .toISOString()
-            .replace(/:..\..*/, ':00.000Z'), // to preserve privacy, intentionally reduce precision of the timestamp
-          app_version: APP_VERSION, // document the app version that was used to process this response
-          country_code: countryCode,
-          postal_code: mapPostalCode(response).postal_code, // to protect the privacy of participants from very small postal code areas, they are merged into larger ones, based on known population data
-          duration: response.duration === null ? null : parseInt(response.duration),
-          abuse_score,
-        };
-        const model: BackendResponseModelT = { ...meta, ...response, ...meta }; // the double "...meta" is just for vanity: we want the meta-fields to appear first in the JSON representation
-        return assertIs(BackendResponseModel)(model); // ensure we still pass runtime validations as well
-      });
-  });
+  const secretPepperValue = await secretPepper;
+  const { readPromise, writePromise } = performAbuseDetection(dynamoDb, fingerprint, val =>
+    hash(val, secretPepperValue),
+  );
+
+  writePromise // we don't really care about the write operation here - it can finish on its own (we only need to handle its possible failure; if it keeps failing we want to know)
+    .catch(err => console.log(`Error: Couldn't write abuse score for response (caused by\n${err}\n)`));
+
+  const abuse_score = await readPromise // we only care about the read operation
```

Comment:
Here I could've used `let` and `try`/`catch` combo, but I find this a better pattern in this case

Reply:
OK, interesting, I would have assumed `try/catch` would be stylistically preferred, since we're now using as many language constructs as possible.

Reply:
Sounds like a stab 😂

Reply:
But in reality, I do find `try`/`catch` somewhat un-ergonomic and cumbersome plus with the weird and not-well-known semantics of returning something inside `try`/`catch` that also has `finally` block (spoiler, code in `finally` overrules whatever you tried to return before).

Reply:
Not taking a jab at anything, promise! :smile: And that pun was 100% un-intentional...

Reply:
>somewhat un-ergonomic and cumbersome

Wholly agreed!